cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/utils/ConfigurationUtils.java

```diff
@@ -0,0 +1,240 @@
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedList;
+import java.util.Properties;
+import java.util.Queue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.devonfw.cobigen.cli.CobiGenCLI;
+import com.devonfw.cobigen.cli.constants.MessagesConstants;
+
+/**
+ *
+ */
+public class ConfigurationUtils {
+
+    /**
+     * Name of templates folder
+     */
+    public static final String COBIGEN_TEMPLATES = "CobiGen_Templates";
+
+    /**
+     * Name of configuration file
+     */
+    private static final String COBIGEN_CONFIG = "config.txt";
+
+    /**
+     * Name of configuration key for custom templates file path
+     */
+    private static final String COBIGEN_CONFIG_LOCATION_KEY = "cobigen.custom-templates-location";
+
+    /**
+     * Logger to output useful information to the user
+     */
+    private static Logger logger = LoggerFactory.getLogger(CobiGenCLI.class);
+
+    /**
+     * Checks if the configuration file exists and returns the path of the custom templates location key
+     * @return Path of custom templates location or null
+     */
+    public Path getCustomTemplatesLocation() {
+        if (Files.exists(Paths.get(getCobigenCliRootPath() + File.separator + COBIGEN_CONFIG))) {
+            Properties props = readConfigFileProperties();
+            return Paths.get(props.getProperty(COBIGEN_CONFIG_LOCATION_KEY));
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * @return Path of Cobigen CLI root
+     */
+    public Path getCobigenCliRootPath() {
+        Path rootCLIPath = null;
+        try {
+            File locationCLI = new File(CobiGenUtils.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+            rootCLIPath = locationCLI.getParentFile().toPath();
+        } catch (URISyntaxException e) {
+            logger.error("An error occured while building the URI of the CLI location {}", e);
+        }
+        return rootCLIPath;
+    }
+
+    /**
+     * Checks if custom templates location path exists and either returns the path of the default location
+     * next to the CLI or the custom templates location defined in the configuration
+     * @return File of Cobigen templates folder
+     */
+    public File getCobigenTemplatesFolderFile() {
+        Path pathForCobigenTemplates;
+
+        Path customTemplatesLocation = getCustomTemplatesLocation();
+        if (customTemplatesLocation != null) {
+            pathForCobigenTemplates = getCustomTemplatesLocation();
+        } else {
+            pathForCobigenTemplates = getCobigenCliRootPath();
+        }
+
+        File cobigenTemplatesFolder = null;
+        if (pathForCobigenTemplates != null) {
+            cobigenTemplatesFolder =
+                pathForCobigenTemplates.resolve(pathForCobigenTemplates + File.separator + COBIGEN_TEMPLATES).toFile();
```

Comment:
you can concatenate .resolve(...).resolve(...)

Reply:
adjusted.